projects/knora-ui/src/lib/viewer/values/color-value/color-picker/color-picker.component.ts

```diff
@@ -0,0 +1,215 @@
+import { FocusMonitor } from '@angular/cdk/a11y';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { Component, DoCheck, ElementRef, HostBinding, Input, OnDestroy, Optional, Self, OnInit } from '@angular/core';
+import { ControlValueAccessor, FormBuilder, FormControl, FormGroup, FormGroupDirective, NgControl, NgForm, Validators } from '@angular/forms';
+import { ErrorStateMatcher, MatFormFieldControl, mixinErrorState, CanUpdateErrorState, CanUpdateErrorStateCtor } from '@angular/material';
+import { Subject } from 'rxjs';
+import { CustomRegex } from '../../custom-regex';
+
+/**
+ * Represents a DateTime consisting of a date and a time.
+ */
+export class ColorPicker {
```

Comment:
Do we actually need an object `ColorPicker`? Couldn't this just be string? This would simplify the handling in the parent component.

Reply:
If it was only a string, then it would work like other components for primitive values.

Reply:
I removed the class and now work with a string.